pkg/securitycontext/util.go

```diff
@@ -63,6 +63,11 @@ func DetermineEffectiveSecurityContext(pod *v1.Pod, container *v1.Container) *v1
 		*effectiveSc.SELinuxOptions = *containerSc.SELinuxOptions
 	}
 
+	if containerSc.WindowsOptions != nil {
+		effectiveSc.WindowsOptions = new(v1.WindowsSecurityContextOptions)
+		*effectiveSc.WindowsOptions = *containerSc.WindowsOptions
```

Comment:
do you expect the container windowsOptions to fully replace the pod-level windowsOptions, or if both are non-nil, should only the non-nil container windowsOptions fields overwrite the pod-level ones?

Reply:
I think it makes more sense to fully replace the pod-level one; it makes little sense to set GMSA names and contents in 2 different places - although arguably this might change as we add more options. Besides, that's the current behaviour for eg SELinuxOptions, seems better to keep the same behaviour?

Reply:
will that still be true as future fields are added to `windowsOptions` (e.g. runAsUsername)?

Reply:
Fair enough, changing this.

Reply:
I could see both gmsa fields getting overridden together (if either is non-nil in the container securityContext, both get used)

Reply:
Done